### ISSUE TITLE
Codecov doesn't work

### ISSUE THREAD

**Issue by 3flex** [arturbosch/detekt#1754]
## Expected Behavior

Codecov works

## Observed Behavior

Codecov doesn't work

## Steps to Reproduce

https://codecov.io/gh/arturbosch/detekt/branch/master/commits

## Context

I've tried many variations to try and get Codecov working reliably. It still doesn't. This makes me sad. If someone can figure it out, I would be very appreciative! Because I don't know what else to try.

## Your Environment

* Version of detekt used:
* Version of Gradle used (if applicable):
* Operating System and version:
* Link to your project (if it's a public repository):

**Comment by schalkms** [comment]
How did you verify this? I couldn't see any problems in the last few PRs.

**Comment by 3flex** [comment]
Click the link in the PR, see "Error" for the last two commits on master branch. #1751 and #1752 have no Codecov reports.

**Comment by schalkms** [comment]
You are right. I haven't paid attention to the actual execution of codecov.
Unfortunately, the error doesn't give any hints on what is wrong.
I'm not even sure if our configuration is somehow wrong or if something with codecovs infrastructure is wrong.

**Comment by schalkms** [comment]
@3flex Does it make sense to revert #1739 to have at least a working test coverage?

**Comment by 3flex** [comment]
I don't think so, coverage reports didn't work before that PR and still don't work now, so reverting will just go back to a different broken state.

**Comment by schalkms** [comment]
Okay. 
I saw uploaded coverage reports when browsing through the commits, which were created before the linked PR.
For example, [this commit](https://codecov.io/gh/arturbosch/detekt/commit/3eb7d4a2bcdae4e899519601fcaad862c787b271) shows a report. It's also integrated into GitHub's PR view.
That's the main reason why I wanted to revert to the old approach, although not everything is apparently working fine.
Here we have at least a helping hand in terms of code coverage.

Do you remember the problem with this approach? (in case someone wants to pick this up)

**Comment by 3flex** [comment]
As far as I know every config I've tried has ended in eventual failure, but I don't know why.

Hopefully someone can contribute to a fix, but I won't be spending any more time on it.

**Comment by schalkms** [comment]
It looks like codecov is back again.
I observed that the report doesn't work when no tracked source files were changed in the PR.
For instance, one of my PRs only changed the `Contributing.md` file. Codecov hasn't created a report for whatever reason.

I also investigated [the sample project](https://github.com/codecov/example-gradle) for configuring codecov as a Gradle task.
I could only find one difference.
They created a separate Gradle task and only run this task when the Travis build completed successfully, whereas we run this task on every build.
Furthermore, they added the `./gradlew codeCoverage` instruction in the `travis.yml` file.
